network-config-analyzer/SchemeRunner.py

```diff
@@ -87,9 +88,13 @@ def _add_config(self, config_entry, peer_container_global):
                 ns_list = self.scheme.get('namespaceList', 'k8s')
             if not pod_list:  # use global resource file
                 pod_list = self.scheme.get('podList', 'k8s')
+
             pod_list = self._handle_resources_list(pod_list)
             ns_list = self._handle_resources_list(ns_list)
             peer_container = PeerContainer(ns_list, pod_list, config_name)
+            # look for service resources under 'pod_list' files
+            services = K8sServiceYamlParser.parse_service_resources(pod_list, peer_container)
```

Comment:
It would create circular dependencies: K8sServiceYamlParser imports PeerContainer.
That's why I put service parser outside of PeerContainer, unlike ns and pod resource parser. Ideally, I think it would be more clean if ns resources and pod resources were parsed in K8sYamlParser or in their separate parsers.

Reply:
So the current situation is that every time right after creating a `PeerContainer` object, you must also write the following code:
```
services = K8sServiceYamlParser.parse_service_resources(pod_list, peer_container)
peer_container.set_services(services)
```
This is currently only in `SchemeRunner`, but would have to be added also for `nca.py` where `PeerContainer` objects are also created (for cli queries).
It creates some code duplication + error-prone, so maybe we should break the dependency of `K8sServiceYamlParser` on `PeerContainer`. 
It seems that the dependency is originated from having the service yaml parser to assign the target pods for the service, and then for checking the named ports of the target pods.
One option is to take this functionality outside to a separate function to be run later after parsing all the services by the `PeerContainer`.

Reply:
I agree with @adisos - `PeerContainer` should call `K8sServiceYamlParser`. Services are a bit more complex compared to Deployments and ServiceAccounts, but we should still try to hide this complexity from the users of `PeerContainer`.

As to the dependency problem, there are two places where `K8sServiceYamlParser` depends on `PeerContainer`:
1. It needs to call `PeerContainer.locate_kube_config_file()` before calling `CmdlineRunner` - this is really an old design bug. There is no reason for this function to reside inside `PeerContainer` - it should move to `CmdlineRunner` and always get called by `CmdlineRunner.get_k8s_resources()`.
2. It needs to set the set of pods behind the service by calling `peer_container.get_peers_with_label()`. This can be done by `PeerContainer` itself after calling the parser, using the key-values stored in `K8sService.selector`. Not extremely elegant, but can work 🤫